addOns/pscan/src/main/java/org/zaproxy/addon/pscan/internal/AddOnScanRulesLoader.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.pscan.internal;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.control.AddOn;
+import org.zaproxy.zap.control.AddOn.InstallationStatus;
+import org.zaproxy.zap.control.ExtensionFactory;
+import org.zaproxy.zap.extension.AddOnInstallationStatusListener;
+import org.zaproxy.zap.extension.AddOnInstallationStatusListener.StatusUpdate;
+import org.zaproxy.zap.extension.pscan.ExtensionPassiveScan;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class AddOnScanRulesLoader implements AddOnInstallationStatusListener {
+
+    private static final Logger LOGGER = LogManager.getLogger(AddOnScanRulesLoader.class);
+
+    private final Map<AddOn, List<PluginPassiveScanner>> addOnScanRules;
+    private final ExtensionPassiveScan extension;
+
+    public AddOnScanRulesLoader(ExtensionPassiveScan extension) {
+        addOnScanRules = new HashMap<>();
+        this.extension = extension;
+    }
+
+    public void load() {
+        ExtensionFactory.getAddOnLoader().getAddOnCollection().getAddOns().stream()
+                .filter(addOn -> addOn.getInstallationStatus() == InstallationStatus.INSTALLED)
+                .forEach(this::loadScanRules);
+    }
+
+    public void unload() {
+        addOnScanRules.keySet().forEach(this::removeScanRules);
+        addOnScanRules.clear();
```

Comment:
Is synchronization needed on `addOnScanRules` for this operation?

Reply:
There are no concurrency, the add-ons are un/installed one at a time.